cirq-web/cirq_ts/src/bloch_sphere/components/vector.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ArrowHelper, Vector3, Group, LineBasicMaterial} from 'three';
+
+/**
+ * Generates a vector to add to the Bloch sphere. The length and direction of the vector are configurable.
+ */
+export class Vector extends Group {
+  readonly length: number;
+  readonly x: number;
+  readonly y: number;
+  readonly z: number;
+
+  /**
+   * Class constructor.
+   * @param vectorData A JSON string containing information used to build the vector.
+   * @returns An instance of the class containing the generated vector. This can be
+   * added to the Bloch sphere instance as well as the scene.
+   */
+  constructor(vectorData?: string) {
+    super();
+
+    if (vectorData) {
+      const parsedObj = JSON.parse(vectorData);
+      this.x = parsedObj.x;
+      this.y = parsedObj.y;
+      this.z = parsedObj.z;
+      this.length = parsedObj.length;
+    } else {
+      this.x = 0;
+      this.y = 0;
+      this.z = 0;
```

Comment:
Recommend _not_ using (0, 0, 0) as the default vector. The direction of the zero-vector is undefined, which means this depends on implicit behavior in three.js and may cause confusion for devs.

Reply:
Good point..fixed